src/dbus_nm.rs

```diff
@@ -0,0 +1,412 @@
+        try!(self.dbus.call(path, NM_DEVICE_INTERFACE, "Disconnect"));
+
+        Ok(())
+    }
+
+    pub fn get_device_access_points(&self, path: &str) -> Result<Vec<String>, String> {
+        self.dbus
+            .property(path, NM_WIRELESS_INTERFACE, "AccessPoints")
+    }
+
+    pub fn get_access_point_ssid(&self, path: &str) -> Option<String> {
+        if let Ok(ssid_vec) = self.dbus.property(path, NM_ACCESS_POINT_INTERFACE, "Ssid") {
+            utf8_vec_u8_to_string(ssid_vec).ok()
+        } else {
+            None
+        }
+    }
+
+    pub fn get_access_point_strength(&self, path: &str) -> Result<u32, String> {
+        self.dbus
+            .property(path, NM_ACCESS_POINT_INTERFACE, "Strength")
+    }
+
+    pub fn get_access_point_flags(&self, path: &str) -> Result<NM80211ApFlags, String> {
+        self.dbus.property(path, NM_ACCESS_POINT_INTERFACE, "Flags")
+    }
+
+    pub fn get_access_point_wpa_flags(&self, path: &str) -> Result<NM80211ApSecurityFlags, String> {
+        self.dbus
+            .property(path, NM_ACCESS_POINT_INTERFACE, "WpaFlags")
+    }
+
+    pub fn get_access_point_rsn_flags(&self, path: &str) -> Result<NM80211ApSecurityFlags, String> {
+        self.dbus
+            .property(path, NM_ACCESS_POINT_INTERFACE, "RsnFlags")
+    }
+}
+
+
+impl VariantTo<DeviceType> for DBusApi {
+    fn variant_to(value: Variant<Box<RefArg>>) -> Option<DeviceType> {
+        variant_to_device_type(value)
+    }
+}
+
+
+impl VariantTo<DeviceState> for DBusApi {
+    fn variant_to(value: Variant<Box<RefArg>>) -> Option<DeviceState> {
+        variant_to_device_state(value)
+    }
+}
+
+
+impl VariantTo<NM80211ApFlags> for DBusApi {
+    fn variant_to(value: Variant<Box<RefArg>>) -> Option<NM80211ApFlags> {
+        variant_to_ap_flags(value)
+    }
+}
+
+
+impl VariantTo<NM80211ApSecurityFlags> for DBusApi {
+    fn variant_to(value: Variant<Box<RefArg>>) -> Option<NM80211ApSecurityFlags> {
+        variant_to_ap_security_flags(value)
+    }
+}
+
+
+fn variant_to_device_type(value: Variant<Box<RefArg>>) -> Option<DeviceType> {
+    value.0.as_i64().and_then(|v| Some(DeviceType::from(v)))
+}
+
+
+fn variant_to_device_state(value: Variant<Box<RefArg>>) -> Option<DeviceState> {
+    value.0.as_i64().and_then(|v| Some(DeviceState::from(v)))
+}
+
+
+fn variant_to_ap_flags(value: Variant<Box<RefArg>>) -> Option<NM80211ApFlags> {
+    value
+        .0
+        .as_i64()
+        .and_then(|v| NM80211ApFlags::from_bits(v as u32))
+}
+
+
+fn variant_to_ap_security_flags(value: Variant<Box<RefArg>>) -> Option<NM80211ApSecurityFlags> {
+    value
+        .0
+        .as_i64()
+        .and_then(|v| NM80211ApSecurityFlags::from_bits(v as u32))
+}
+
+
+pub fn add_val<T>(map: &mut SettingsMap, key: &str, value: T)
+    where T: RefArg + 'static
+{
+    map.insert(key.to_string(), Variant(Box::new(value)));
+}
+
+pub fn add_str(map: &mut SettingsMap, key: &str, value: &str) {
```

Comment:
I'd take `key` and `value` (and `key` above) as type parameters of `T: Into<String>, U: Into<String>`. This lets the callsite use string literals if they want, but also not copy a `String` if they already have one...